### PR TITLE
issue-3290: test class deprecated asList methods  fix in IngressEnri…

### LINKED ISSUE #3290: IngressEnricherTest : Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Task description

In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/e1cde0bd0d4ab1a659db7b1f7a884705048b27de/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java#L119

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [x] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by vijaybhagwat24**
## Description

Fixes #3290

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x ] My code follows the style guidelines of this project
 - [x ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3321 (_2024-08-16T05:07:06Z_)

:gear: [JKube E2E Tests (10414964407)](https://github.com/eclipse-jkube/ci/actions/runs/10414964407) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3321)](https://github.com/eclipse-jkube/ci/actions/runs/10414964407/job/28844746028)

**Comment by vijaybhagwat24** [comment]
test cases run at local:
image

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java [hunk 1]
```diff
@@ -210,7 +213,7 @@ void getIngressTlsXMLConfig_withNonNullResourceConfig() {
         List<IngressTlsConfig> ingressTlsConfigs = IngressEnricher.getIngressTlsXMLConfig(resourceConfig);
 
         // Then
-        assertThat(ingressTlsConfigs).asList().hasSize(1);
+        assertThat(ingressTlsConfigs).asInstanceOf(InstanceOfAssertFactories.list(IngressTlsConfig.class)).hasSize(1);
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:216]: I think `asList()` call is redundant here. Could you please check if it works without `asList()` ?

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java [hunk 2]
```diff
@@ -219,7 +222,7 @@ void getIngressTlsXMLConfig_withNullResourceConfig() {
         List<IngressTlsConfig> ingressTlsConfigs = IngressEnricher.getIngressTlsXMLConfig(null);
 
         // Then
-        assertThat(ingressTlsConfigs).asList().isEmpty();
+        assertThat(ingressTlsConfigs).asInstanceOf(InstanceOfAssertFactories.list(IngressTlsConfig.class)).isEmpty();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:225]: https://github.com/eclipse-jkube/jkube/pull/3321/files#r1719649078

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java [hunk 3]
```diff
@@ -192,7 +195,7 @@ void getIngressRuleXMLConfig_withNullResourceConfig() {
         List<IngressRuleConfig> ingressRuleConfigs = IngressEnricher.getIngressRuleXMLConfig(null);
 
         // Then
-        assertThat(ingressRuleConfigs).asList().isEmpty();
+        assertThat(ingressRuleConfigs).asInstanceOf(InstanceOfAssertFactories.list(IngressRuleConfig.class)).isEmpty();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:198]: https://github.com/eclipse-jkube/jkube/pull/3321/files#r1719649078

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java [hunk 4]
```diff
@@ -183,7 +186,7 @@ void getIngressRuleXMLConfig_withNonNullResourceConfig() {
         List<IngressRuleConfig> ingressRuleXMLConfig = IngressEnricher.getIngressRuleXMLConfig(resourceConfig);
 
         // Then
-        assertThat(ingressRuleXMLConfig).asList().hasSize(1);
+        assertThat(ingressRuleXMLConfig).asInstanceOf(InstanceOfAssertFactories.list(IngressRuleConfig.class)).hasSize(1);
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:189]: https://github.com/eclipse-jkube/jkube/pull/3321/files#r1719649078

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java [hunk 5]
```diff
@@ -110,27 +113,27 @@ void createIngressFromXMLConfigWithConfiguredServiceName() {
...
         assertThat(kubernetesListBuilder)
-            .extracting(KubernetesListBuilder::buildItems).asList()
+            .extracting(KubernetesListBuilder::buildItems).asInstanceOf(InstanceOfAssertFactories.list(HasMetadata.class))
             .hasSize(2)
             .element(1).asInstanceOf(InstanceOfAssertFactories.type(Ingress.class))
             .hasFieldOrPropertyWithValue("apiVersion", "networking.k8s.io/v1")
             .hasFieldOrPropertyWithValue("metadata.name", providedService.getMetadata().getName())
             .extracting(Ingress::getSpec)
-            .satisfies(is -> assertThat(is).extracting("tls").asList().element(0)
+            .satisfies(is -> assertThat(is).extracting("tls").asInstanceOf(InstanceOfAssertFactories.list(IngressSpec.class)).element(0)
```
[the remark this answers is not in the document]
Comment (vijaybhagwat24) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:122]: Modiying this one

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java (hunk 1 above)
Comment (vijaybhagwat24) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:216]: agree, tested working fine. I will modify this.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java (hunk 2 above)
Comment (vijaybhagwat24) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:225]: yes, agree. I will modify this

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java (hunk 3 above)
Comment (vijaybhagwat24) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:198]: I will modify.

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java (hunk 4 above)
Comment (vijaybhagwat24) [reply jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/IngressEnricherTest.java:189]: i will modify.

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**